### ISSUE TITLE
Doubling group changes  

### ISSUE THREAD

**Issue by brun0xon** [rodydavis/undo#1]
Hello. There is a bag:
My situation:
```
    var changes = ChangeStack();
    var a = 0;
    var change = Change.inline(() => a++, () => a--);

    changes.group();
    changes.add(change);
    print(a); // 1
    changes.add(change);
    print(a); // 2
    changes.commit();
    print(a); // 4
```
Situation in your example:
```
count = 0;

changes.group();

var increase = new Change.inline(() => count++, () => count--);
changes.add(increase);
changes.add(increase);

changes.commit();
print(count); // 2

changes.undo();
print(count); // 0
```
Plugin: undo 1.1.0+1

**Comment by rodydavis** [comment]
Fixed on 1.2.0, Refactored and should work as expected. There is a new interface but I updated the example and should be easier to use

**Comment by brun0xon** [comment]
Good!
